Reset selection on DataProvider change in RadioButtonGroup

anasmi opened the pull request:
Fixes #11454

Other components needing the same fix are ListSelect,NativeSelect,Combobox (The ones implementing HasDataProvider and extending AbstractSingleSelect. Grid is not affected as it extends AbstractListing and hadles selection differently)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/11526)
<!-- Reviewable:end -->

Commit:
Reset selection on DataProvider change in RadioButtonGroup

Fixes #11454

Other components needing the same fix are ListSelect,NativeSelect,Combobox (The ones implementing HasDataProvider and extending AbstractSingleSelect. Grid is not affected as it extends AbstractListing and hadles selection differently)

pleku reviewed (changes requested):
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11526)* status: all files reviewed, 1 unresolved discussion (waiting on @anasmi)

___
*[uitest/src/test/java/com/vaadin/tests/components/radiobuttongroup/RadioButtonGroupChangeDataProviderTest.java, line 16 at r1](https://reviewable.io/reviews/vaadin/framework/11526#-LbPPfMrC8hWm8JxZC6e:-LbPPfMrC8hWm8JxZC6f:byusoxg) ([raw file](https://github.com/vaadin/framework/blob/004a646c201f5a2802be45e9bebdc15fc6ee8781/uitest/src/test/java/com/vaadin/tests/components/radiobuttongroup/RadioButtonGroupChangeDataProviderTest.java#L16)):*
> ```
> verifyOptionIsSelectable() {
> ```
You should also verify that the value change event from old value to new value `null` is fired when changing the data provider


<!-- Sent from Reviewable.io -->

Commit:
Added test case

Verify that ValueChangeListener is actually fired after value is reset setting new DataProvider

anasmi commented:
___
*[uitest/src/test/java/com/vaadin/tests/components/radiobuttongroup/RadioButtonGroupChangeDataProviderTest.java, line 16 at r1](https://reviewable.io/reviews/vaadin/framework/11526#-LbPPfMrC8hWm8JxZC6e:-LbRpFe4EyEoyKcR1Pjm:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/004a646c201f5a2802be45e9bebdc15fc6ee8781/uitest/src/test/java/com/vaadin/tests/components/radiobuttongroup/RadioButtonGroupChangeDataProviderTest.java#L16)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

> ```
> verifyOptionIsSelectable() {
> ```
You should also verify that the value change event from old value to new value `null` is fired when changing the data provider
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

pleku reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11526#-:-LbT1wrr-TVjGfPKBFlR:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

Commit:
Merge branch 'master' into #11454

ZheSun88 committed:
Merge branch 'master' into #11454
